### PR TITLE
[DEV-9633] Use latest_transaction_search instead of latest_transaction

### PR DESCRIPTION
**Pull request by aguest-kc**
**Description:**
Use the `latest_transaction_search` field instead of the `latest_transaction` field in spending_filter to reduce the number of joins to the transaction_search table.

**Technical details:**
Use the `latest_transaction_search` field instead of the `latest_transaction` field in spending_filter to reduce the number of joins to the transaction_search table. Previously, this filter was joining to the `transaction_search` table multiples times through different views and this was causing poor performance. This change uses the `transaction_search` table directly to reduce those JOINs to just one.

**Requirements for PR merge:**

3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed
6. [x] Data validation completed
8. [x] Jira Ticket [DEV-9633](https://federal-spending-transparency.atlassian.net/browse/DEV-9633):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
1. Unit & integration tests updated
No new tests are needed for this change since we're still using the same table for the same data, just reducing the number of JOIN statements.

2. API documentation updated
No API documentation is affected by this change.

5. Frontend impact assessment completed
The frontend is not affected by this change.

7. Appropriate Operations ticket(s) created
No operations tickets are needed for this change.
```

### CODE REVIEW

File: usaspending_api/spending_explorer/v2/filters/spending_filter.py [hunk 1]
```diff
@@ -64,10 +64,7 @@ def spending_filter(alt_set, queryset, filters, _type):
 
             # recipient
             elif key == "recipient":
-                alt_set = alt_set.filter(
-                    Q(award__latest_transaction__assistance_data__awardee_or_recipient_legal=value)
-                    | Q(award__latest_transaction__contract_data__awardee_or_recipient_legal=value)
-                )
+                alt_set = alt_set.filter(Q(award__latest_transaction_search__recipient_name_raw=value))
```
Comment (dporth-frb) [inline usaspending_api/spending_explorer/v2/filters/spending_filter.py:67]: As your story told you to do, I see you updated to `latest_transaction_search`, but why did you change this query from what it used to do (beyond the `latest_transaction_search` update)?

File: usaspending_api/spending_explorer/v2/filters/spending_filter.py [hunk 2]
```diff
@@ -106,8 +103,7 @@ def spending_filter(alt_set, queryset, filters, _type):
             elif key == "recipient":
                 queryset = queryset.filter(
                     treasury_account__in=alt_set.filter(
-                        Q(award__latest_transaction__contract_data__awardee_or_recipient_legal=value)
-                        | Q(award__latest_transaction__assistance_data__awardee_or_recipient_legal=value)
+                        Q(award__latest_transaction_search__recipient_name_raw=value)
```
Comment (dporth-frb) [inline usaspending_api/spending_explorer/v2/filters/spending_filter.py:106]: Same question here.

File: usaspending_api/spending_explorer/v2/filters/spending_filter.py (hunk 1 above)
Comment (aguest-kc) [reply usaspending_api/spending_explorer/v2/filters/spending_filter.py:67]: `latest_transaction__assistance_data` and `latest_transaction__contract_data` are both views on top of `transaction_search`. One is a view where `is_fpds = False` and the other view where `is_fpds = True`. Since the `is_fpds` column isn't null-able, there's no reason to join to the `transaction_search` twice like we were before.

Comment (collinwr) [inline usaspending_api/spending_explorer/v2/filters/spending_filter.py:67]: I believe the existing columns (`awardee_or_recipient_legal`), on the two tables, are currently mapped to `recipient_name` not `recipient_name_raw`

Comment (aguest-kc) [reply usaspending_api/spending_explorer/v2/filters/spending_filter.py:67]: It looks like you're correct. I just used `recipient_name_raw` since that's the column that the ticket mentioned. I'll update this query to use recipient name.

This is from the FPDS to Transaction Search mappings:
```jsonc
# transaction_fpds col name : transaction_search col name
"awardee_or_recipient_legal": "recipient_name
```

Comment (aguest-kc) [reply usaspending_api/spending_explorer/v2/filters/spending_filter.py]: Updated the recipient column

Comment (collinwr) [reply usaspending_api/spending_explorer/v2/filters/spending_filter.py]: Thanks Andy! I didn't see that the ticket mentioned the `_raw` field.

**Review by collinwr — APPROVED** [review]
